subsys/net/l2/ethernet/arp.c

```diff
@@ -377,8 +377,9 @@ struct net_pkt *net_arp_prepare(struct net_pkt *pkt,
 		if (ipv4) {
 			addr = &ipv4->gw;
 			if (net_ipv4_is_addr_unspecified(addr)) {
-				NET_ERR("Gateway not set for iface %p",
-					net_pkt_iface(pkt));
+				NET_ERR("Gateway not set for iface %p, could not "
```

Comment:
While we are at it, lets print interface index instead of the pointer value. I have been changing these in various PRs, the pointer value is not very useful for the interface.
So `... iface %d, ....", net_if_get_by_iface(net_pkt_iface(pkt)), ...`